### ISSUE TITLE
Dependency Issues HPC Environment

### ISSUE THREAD

**Issue by jayacharan4** [netdata/netdata#6859]
##### Question summary
I faced a problem when I tried to install netdata with all plugins with curl and I have had the dependency issues, it was not a problem when I did basic install through github. Now that I want all plugins I tried to curl as mentioned in the documentation, I have a libcom_err now :
"Error: Package: libcom_err-devel-1.42.9-13.el7.x86_64 (base)
           Requires: libcom_err(x86-64) = 1.42.9-13.el7
           Installed: libcom_err-1.44.3.wc1-0.el7.x86_64 (installed)
               libcom_err(x86-64) = 1.44.3.wc1-0.el7
           Available: libcom_err-1.42.9-13.el7.x86_64 (base)
               libcom_err(x86-64) = 1.42.9-13.el7
 You could try using --skip-broken to work around the problem" 
I tried many ways to fix this but does not work out. I am looking to install netdata in a HPC Environment.
##### OS / Environment
CentOS 7
##### Component Name
libcom_err
##### Expected results
I am expecting to install netdata with all the plugins on my system.

**Comment by paulkatsoulakis** [comment]
Hello @jayacharan4, thanks for reaching out. I am sorry you faced troubles during install.
I am afraid i am going to need more information from you, please provide the following:

1) `cat /etc/os-release` from the system you are trying to install netdata
2) please provide the output of `uname -a` and `uname -m`

3) Please provide the exact command that you run and the full output that the command produced. Feel free to attach it with a fail if the text is too long. Also make sure you remove any sensitive information. 

To explain a bit further, the errors you provided are failures on yum packaging tool of your system (centos uses yum) saying that a dependency has failed, but you haven't provided the full output to see which package raised this dependency. Are you sure this is netdata related?

**Comment by jayacharan4** [comment]
1.
NAME="CentOS Linux"
VERSION="7 (Core)"
 ID="centos"
ID_LIKE="rhel fedora"
VERSION_ID="7"

2.
 **genius.hpcc.ttu.edu 3.10.0-862.14.4.el7.x86_64 #1 SMP Wed Sep 26 15:12:11 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux**
**x86_64**

3. 
curl -Ss 'https://raw.githubusercontent.com/netdata/netdata-demo-site/master/install-required-packages.sh' >/tmp/kickstart.sh && bash /tmp/kickstart.sh -i netdata-all 


So I tried running the above command it, it did not work. However I tried installing netdata through github it worked well. But when I had to do it through one line install it doesn't work out. I am also trying out a way to also install a database package through github. For me it doesn't work out through curl. So the problem I saw was with yum repository.

Error:

--> Running transaction check
---> Package glibc.i686 0:2.17-260.el7 will be updated
--> Processing Dependency: glibc = 2.17-260.el7 for package: glibc-common-2.17-260.el7.x86_64
---> Package glibc.x86_64 0:2.17-260.el7 will be updated
--> Processing Dependency: glibc = 2.17-260.el7 for package: glibc-common-2.17-260.el7.x86_64
---> Package libXau-devel.x86_64 0:1.0.8-2.1.el7 will be installed
--> Finished Dependency Resolution
Error: Package: glibc-common-2.17-260.el7.x86_64 (@base)
           Requires: glibc = 2.17-260.el7
           Removing: glibc-2.17-260.el7.i686 (@base)
               glibc = 2.17-260.el7
           Updated By: glibc-2.17-260.el7_6.6.i686 (updates)
               glibc = 2.17-260.el7_6.6
           Available: glibc-2.17-260.el7_6.3.i686 (updates)
               glibc = 2.17-260.el7_6.3
           Available: glibc-2.17-260.el7_6.4.i686 (updates)
               glibc = 2.17-260.el7_6.4
           Available: glibc-2.17-260.el7_6.5.i686 (updates)
               glibc = 2.17-260.el7_6.5
 You could try using --skip-broken to work around the problem


I don't know if the i686 is affecting that.

--> Processing Dependency: libcom_err(x86-64) = 1.42.9-13.el7 for package: libcom_err-devel-1.42.9-13.el7.x86_64
---> Package libkadm5.x86_64 0:1.15.1-37.el7_6 will be installed
---> Package libverto-devel.x86_64 0:0.2.5-4.el7 will be installed
---> Package openssl.x86_64 1:1.0.2k-16.el7 will be updated
---> Package openssl.x86_64 1:1.0.2k-16.el7_6.1 will be an update
--> Finished Dependency Resolution
Error: Package: libcom_err-devel-1.42.9-13.el7.x86_64 (base)
           Requires: libcom_err(x86-64) = 1.42.9-13.el7
           Installed: libcom_err-1.44.3.wc1-0.el7.x86_64 (installed)
               libcom_err(x86-64) = 1.44.3.wc1-0.el7
           Available: libcom_err-1.42.9-13.el7.x86_64 (base)
               libcom_err(x86-64) = 1.42.9-13.el7
 You could try using --skip-broken to work around the problem
** Found 2 pre-existing rpmdb problem(s), 'yum check' output follows:
glibc-common-2.17-260.el7_6.6.x86_64 is a duplicate with glibc-common-2.17-260.el7.x86_64
glibc-common-2.17-260.el7_6.6.x86_64 has missing requires of glibc = ('0', '2.17', '260.el7_6.6')

**Comment by Saruspete** [comment]
Hello,
It seems you are using a `libcom_err` package that is newer (1.44) than the one officially provided by CentOS (1.42)
Seeing the "wc", it's from [WhamCloud](https://downloads.whamcloud.com/public/e2fsprogs/1.44.3.wc1/el7/RPMS/x86_64/), most likely because you have Lustre FS (btw, did you try BeeGFS ?). From the repo content, they broke the 32/64 multilib policy: they only provided the 64 bits version (x86_64) while they also should have provided the 32 (i686). 
As Yum will try to have the same version of both libraries, that will generate you issues with other packages too.
The gore fix for that is to ignore the multilib dependency with `yum --setopt=protected_multilib=false install ...` but that's not a good policy: you should request WC to do things well and provide 32bits.

So, just to fix the netdata issue, as you can't downgrade the package version, you should rebuild a custom RPM for your system.
To do that, on your "build" or "admin" server, issue the following :
```
# As root, to install required package
sudo yum install rpm-build

# As standard (non-root) user:
git clone https://github.com/netdata/netdata.git
cd netdata
git checkout $(cat packaging/version)
contrib/rhel/build-netdata-rpm.sh
```

That will generate packages in `$HOME/rpmbuild/RPMS/x86_64/netdata-v1.17.1.*.rpm`, hopefully with your custom version of glibc & libcom_err.
